### PR TITLE
add baidu vectordb as vector store

### PR DESCRIPTION
**Pull request by fengjial**
# Description

This PR add baidu vector db as vector sotre

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+        self,
+        dimension: int,
+        table_name: str = DEFAULT_TABLE_NAME,
+        replication: int = DEFAULT_REPLICA,
+        partition: int = DEFAULT_PARTITION,
+        index_type: str = DEFAULT_INDEX_TYPE,
+        metric_type: str = DEFAULT_METRIC_TYPE,
+        drop_exists: Optional[bool] = False,
+        vector_params: Optional[Dict] = None,
+        filter_fields: Optional[List[TableField]] = [],
```
Comment (skvrd) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py:105]: `[]` as a default can produce unexpected behaviors. https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument
```suggestion
        filter_fields: Optional[List[TableField]] = None,
    ):
        if filter_fields is None:
            filter_fields = []
```

Comment (fengjial) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py]: I greatly appreciate your prompt response. The issue has been resolved on my end. Would you be so kind as to check if there are any further comments or concerns? Thank you very much. @skvrd

File: llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+    def _create_database_if_not_exists(self, database_name: str) -> None:
+        db_list = self.vdb_client.list_databases()
+
+        if database_name in [db.database_name for db in db_list]:
+            self.database = self.vdb_client.database(database_name)
+        else:
+            self.database = self.vdb_client.create_database(database_name)
+
+    def _create_table(
+        self, database_name: str, table_params: TableParams
```
Comment (materlai) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py:201]: variable @database_name is never used and can be removed

File: llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+        from pymochow.model.enum import IndexType
+        from pymochow.model.schema import HNSWParams
+
+        vector_params = (
+            {}
+            if table_params.vector_params is None
+            else table_params.vector_params
+        )
+
+        if index_type == IndexType.HNSW:
```
Comment (materlai) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py:272]: how about FLAT, do we need to add support?

**Comment by hatianzhang** [comment]
@fengjial could you help fix the unit test and lint errors

File: llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py (hunk 2 above)
Comment (materlai) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py]: It is resolved and I will delete this comment.

File: llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+            database_name=database_name,
+            table_params=table_params,
+            batch_size=batch_size,
+            **kwargs,
+        )
+
+    def _init_client(self, endpoint: str, account: str, api_key: str) -> None:
+        import pymochow
+        from pymochow.configuration import Configuration
+        from pymochow.auth.bce_credentials import BceCredentials
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py:183]: Can we make sure the pyproject.toml has all the dependencies?

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**